adapters/repos/db/lsmkv/segment_group.go

```diff
@@ -229,6 +229,15 @@ func newSegmentGroup(logger logrus.FieldLogger, metrics *Metrics,
 	}
 
 	for _, entry := range list {
+		if filepath.Ext(entry.Name()) == DeleteMarkerSuffix {
+			// marked for deletion, but never actually deleted. Delete now.
+			if err := os.Remove(filepath.Join(sg.dir, entry.Name())); err != nil {
+				return nil, fmt.Errorf("delete file already marked for deletion %q: %w", entry.Name(), err)
```

Comment:
**question:** do wanna break the loop here or silently log ?

Reply:
Good point, I think I agree with you. The fact that we can't delete an old file (for whatever reason) does not have to block the initialization of a segment. I think logging would be the better alternative than returning.

Reply:
udpated.